File: solutions/exercise-3/src/main/java/io/miragon/training/adapter/inbound/cibseven/SendWelcomeMailDelegate.java
package io.miragon.training.adapter.inbound.cibseven;

import io.miragon.training.application.port.inbound.SendWelcomeMailUseCase;
import io.miragon.training.domain.SubscriptionId;
import org.cibseven.bpm.engine.delegate.DelegateExecution;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class SendWelcomeMailDelegate extends BaseDelegate {

    private final SendWelcomeMailUseCase useCase;

    public SendWelcomeMailDelegate(SendWelcomeMailUseCase useCase) {
        this.useCase = useCase;
    }

    @Override
    protected void executeTask(DelegateExecution execution) {
        var subscriptionId = (String) execution.getVariable("subscriptionId");
        log.debug("Received task to send welcome mail for subscription: {}", subscriptionId);
        useCase.sendWelcomeMail(new SubscriptionId(UUID.fromString(subscriptionId)));
    }
}

File: solutions/exercise-3/src/main/kotlin/io/miragon/training/adapter/inbound/cibseven/SendWelcomeMailDelegate.kt
package io.miragon.training.adapter.inbound.cibseven

import io.miragon.training.application.port.inbound.SendWelcomeMailUseCase
import io.miragon.training.domain.MembershipId
import org.cibseven.bpm.engine.delegate.DelegateExecution
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class SendWelcomeMailDelegate(
    private val useCase: SendWelcomeMailUseCase,
) : BaseDelegate() {

    override fun executeTask(execution: DelegateExecution) {
        val membershipId = execution.getVariable("membershipId") as String
        log.debug { "Received task to send welcome mail for membership: $membershipId" }
        useCase.sendWelcomeMail(MembershipId(UUID.fromString(membershipId)))
    }
}
